modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/portlet/action/EditSEOMVCActionCommand.java

```diff
@@ -98,6 +101,20 @@ protected void doProcessAction(
 			PropertiesParamUtil.getProperties(
 				actionRequest, "TypeSettingsProperties--");
 
+		for (Map.Entry<Locale, String> entry : robotsMap.entrySet()) {
+			String value = entry.getValue();
+
+			if (Validator.isNotNull(value) &&
+				(StringUtil.containsIgnoreCase(
+					value, "nofollow", StringPool.BLANK) ||
+				 StringUtil.containsIgnoreCase(
+					 value, "noindex", StringPool.BLANK))) {
+
+				formTypeSettingsUnicodeProperties.setProperty(
+					LayoutTypePortletConstants.SITEMAP_INCLUDE, "0");
+			}
```

Comment:
Can you break out of the loop here also?

Reply:
Yes, good catch, thank you.